db/migrate/20190425121602_add_racing.rb

```diff
@@ -0,0 +1,81 @@
+class AddRacing < ActiveRecord::Migration[6.0]
+  def change
+    create_table :standard_races do |t|
+      t.belongs_to :category, foreign_key: true, null: false
+      t.belongs_to :user,     foreign_key: true, null: false
+
+      t.string   :status_text, null: false
+      t.boolean  :listed,      null: false, default: true
+      t.boolean  :invite,      null: false, default: false
```

Comment:
Any specific reason for making race publicity two booleans (rather than one enum)?

imo the default way of representing stuff in our database should be equivalent to how we represent it to users, unless there's a good argument not to.

Specifically here, I feel two fields is harder to reason about and needs special methods (the ones on Raceable) to abstract away the difficulty. They have four possible states but we only use three, so it's confusing to figure out which three are valid -- `invite` is in this weird undefined / unused state when `listed` is false. The fields are almost always written together and read together, so they're mostly behaving as if they're one field.

If we ever decide to add a fourth type of publicity (say, "open to my Twitch followers") there's no guarantee it will slot into this system, so we'd need to add another field for that. If we use an enum up front, we can add as many states as we want to later without making things more complicated.

Reply:
I had only briefly seen enums mentioned before in the rails docs, but that seems much better.  I'll take a look at changing that over instead of having a bunch of boolean fields.